spotfinder/kernels/erosion.cu

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file erosion.cu
+ * @brief Contains the CUDA kernel implementation for performing
+ *        morphological erosion on an image using a given kernel radius
+ *        and Chebyshev distance threshold.
+ *
+ * This kernel processes a dispersion mask containing potential signal
+ * spots and background pixels. The kernel processes each pixel in the
+ * mask and iterates over its local neighbourhood defined by a given
+ * kernel radius. The kernel then checks if each pixel is within a given
+ * Chebyshev distance threshold of a background pixel. If the pixel is
+ * within the threshold, it is considered part of the background and is
+ * marked as a background pixel in the erosion mask. Therefore eroding
+ * the edges of the signal spots.
+ * 
+ * @note The __restrict__ keyword is used to indicate to the compiler
+ *       that the two pointers are not aliased, allowing the compiler to
+ *       perform more aggressive optimizations.
+ */
+
+#include <cooperative_groups.h>
+#include <cooperative_groups/reduce.h>
+
+#include "../spotfinder.cuh"
```

Comment:
We believe all cases of this should instead be including common/cuda_common

Reply:
https://github.com/DiamondLightSource/fast-feedback-service/pull/1/commits/b0b1c97d652a3e44ab48ce0743209c435e554363